src/Centreon/Domain/Engine/EngineService.php

```diff
@@ -496,24 +498,112 @@ public function submitServiceResult(SubmitResult $result, Service $service): voi
+        }
+
+        if ($service->getHost() == null) {
+            throw new EngineException(
+                sprintf(_('The host of service (id: %d) is not defined'), $service->getId())
+            );
+        }
+        if (empty($service->getHost()->getName())) {
+            throw new EngineException(
+                sprintf(_('Host name of service (id: %d) can not be empty'), $service->getId())
+            );
+        }
+        if (empty($service->getDescription())) {
+            throw new EngineException(
+                sprintf(_('The description of service (id: %d) can not be empty'), $service->getId())
+            );
+        }
+        $preCommand = sprintf(
+            'ADD_SVC_COMMENT;%s;%s;1;%s;%s',
+            $service->getHost()->getName(),
+            $service->getDescription(),
+            $this->contact->getAlias(),
+            $comment->getComment()
+        );
+        $commandToSend = str_replace(['"', "\n"], ['', '<br/>'], $preCommand);
+        $command = $this->createCommandHeader($service->getHost()->getPollerId(), $comment->getDate()) . $commandToSend;
+
+        $this->engineRepository->sendExternalCommand($command);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function addHostComment(Comment $comment, Host $host): void
+    {
+        // We validate the comment instance
+        $errors = $this->validator->validate(
+            $comment,
+            null,
+            CommentService::VALIDATION_GROUPS_HOST_ADD_COMMENT
+        );
+
+        if ($errors->count() > 0) {
+            throw new ValidationFailedException($errors);
+        }
+
+        if (empty($this->contact->getAlias())) {
+            throw new EngineException(_('The contact alias is empty'));
+        }
+        if ($host == null) {
+            throw new EngineException(_('Host not found'));
+        }
+        if (empty($host->getName())) {
+            throw new EngineException(
+                sprintf(_('Host name can not be empty for host (id: %d)'), $host->getId())
+            );
+        }
+
+        $preCommand = sprintf(
+            'ADD_HOST_COMMENT;%s;1;%s;%s',
+            $host->getName(),
+            $this->contact->getAlias(),
+            $comment->getComment()
+        );
+        $commandToSend = str_replace(['"', "\n"], ['', '<br/>'], $preCommand);
+        $command = $this->createCommandHeader($host->getPollerId(), $comment->getDate()) . $commandToSend;
+        $this->engineRepository->sendExternalCommand($command);
+    }
+
     /**
      * Create the command header for external commands
      *
      * @param int $pollerId Id of the poller
+     * @param \DateTime|null $date date of the command
      * @return string Returns the new generated command header
      * @throws \Exception
      */
-    private function createCommandHeader(int $pollerId): string
+    private function createCommandHeader(int $pollerId, ?\DateTime $date): string
```

Comment:
This way, you maintain compatibility on previous calls.
```suggestion
    private function createCommandHeader(int $pollerId, \DateTime $date = null): string
```